@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -486,14 +470,29 @@ export class PoiRenderer {
      * @param env - TODO! The current zoomLevel level of {@link MapView}
      *
      * @returns `True` if the space is not already allocated by another object (text label or POI)
+     * or a Promise if the image has to be loaded first. This happens when the user explicitly
+     * doesn't load the image when it is put into the cache.
      */
-    prepareRender(pointLabel: TextElement, env: Env): boolean {
+    prepareRender(pointLabel: TextElement, env: Env): boolean | Promise<boolean> {
         const poiInfo = pointLabel.poiInfo;
         if (poiInfo === undefined) {
             return false;
         }
+        const { imageItem, imageTexture } = this.imageCached(poiInfo.imageTextureName);
+        if (!imageItem) {
+            // Set the POI to be invalid, so it will not be rendered, nor will it be re-checked in a
+            // future request. The item must exist (but not necessarily loaded in the cache before
+            // the POI can be initialized). We don't log an error, because the API of the
+            // MapViewImageCache allows the user to delete an image from the cache. The geometry
+            // creation is handled using some task queue which means that in some cases, a POI entry
+            // may be removed before the geometry is created. See queueGeometryCreation, which
```

Comment:
I would avoid to make explicit references to code outside this file that is not even part of the API. They become obsolete quickly.

Reply:
I'll remove it.